executor/mongodb/mongod_test.go

```diff
@@ -65,3 +70,86 @@ func TestExecutorMongoDBMkdir(t *gotesting.T) {
 	}
 	assert.True(t, stat.IsDir())
 }
+
+func TestExecutorMongoDBIsStarted(t *gotesting.T) {
+	assert.False(t, testMongod.IsStarted())
+}
+
+func TestExecutorMongoDBStart(t *gotesting.T) {
+	// get random open TCP port for mongod
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		panic(err)
+	}
+	listenPort := listener.Addr().(*net.TCPAddr).Port
+	listener.Close()
+
+	// get tmpdir for mongod dbPath
+	tmpDBPath, err := ioutil.TempDir("", "TestExecutorMongoDBStartDBPath")
+	assert.NoError(t, err)
+	defer os.RemoveAll(tmpDBPath)
+
+	// get tmpdir
+	tmpPath, err := ioutil.TempDir("", "TestExecutorMongoDBStartTmpPath")
+	assert.NoError(t, err)
+	testConfig.TmpDir = tmpPath
+	defer os.RemoveAll(tmpPath)
+
+	// make the security.keyFile tmpfile
+	tmpKeyFile, err := ioutil.TempFile("", "TestExecutorMongoDBStartKeyFile")
+	assert.NoError(t, err)
+	_, err = tmpKeyFile.Write([]byte("123456789101112"))
+	assert.NoError(t, err)
+	tmpKeyFile.Close()
+	defer os.Remove(tmpKeyFile.Name())
+
+	// make the config tmpdir
+	tmpConfigDir, err := ioutil.TempDir("", "TestExecutorMongoDBStartConfigDir")
+	assert.NoError(t, err)
+	testConfig.ConfigDir = tmpConfigDir
+	defer os.RemoveAll(tmpConfigDir)
+
+	// write mongod config
+	config := &mdbconfig.Config{
+		Net: &mdbconfig.Net{
+			BindIp: "127.0.0.1",
+			Port:   listenPort,
+		},
+		Security: &mdbconfig.Security{
+			KeyFile: tmpKeyFile.Name(),
+		},
+		Storage: &mdbconfig.Storage{
+			DbPath: tmpDBPath,
+		},
+	}
+	assert.NoError(t, config.Write(testConfig.ConfigDir+"/mongod.conf"))
+
+	testMongod = NewMongod(testConfig)
+	assert.NotNil(t, testMongod, ".NewMongod() should not return nil")
+	assert.NoError(t, testMongod.Start(), ".Start() should not return an error")
+
+	var tries int
+	var maxTries int = 60
```

Comment:
should omit type int from declaration of var maxTries; it will be inferred from the right-hand side